### PR TITLE
vkconfig: Beta 4 updates

### PR DESCRIPTION
**Pull request by richard-lunarg**
Change-Id: I023ae6f9654de74b69b66b3b0fb2ac1581d7e067
Cleanup for beta 4
Class member variables are prefixed with _ rather than suffix (discussed with larger LunarG team)
Removed unused test tab from vkvia output dialog.
Settings tree properly saved/restored after configuration edit operations and on program shutdown/restart.
VUID fixes after regression. Still a WIP, but better and works correctly for beta 4.

### CODE REVIEW

**Comment by richard-lunarg** [comment]
Also fixed when closing, don't clear the settings tree if the user decides not to close.
Removed some unused signal/slot connections to eliminate runtime warnings.
These changes were made after beta4 was posted.

File: vkconfig/mainwindow.cpp [hunk 1]
```diff
@@ -759,18 +800,22 @@ void MainWindow::DuplicateClicked(ConfigurationListItem *item) {
...
     (void)item;  // We don't need this
-
+    PushCurrentItem();
     Configurator &configurator = Configurator::Get();
 
     QString full_suggested_path = configurator.GetPath(Configurator::LastImportPath);
     QString full_import_path =
         QFileDialog::getOpenFileName(this, "Import Layers Configuration File", full_suggested_path, "*.json");
-    if (full_import_path.isEmpty()) return;
+    {
```
Comment (lunarpapillo) [inline vkconfig/mainwindow.cpp:809]: Why is this couple of lines in its own block?

File: vkconfig/vuidsearchwidget.cpp [hunk 2]
```diff
@@ -19,72 +19,80 @@
...
+    _user_box->installEventFilter(this);
 
-    search_vuid_ = nullptr;  // Safe to delete a pointer
+    _search_vuid = nullptr;  // Safe to delete a pointer
     ResetCompleter();
 
-    connect(add_button_, SIGNAL(pressed()), this, SLOT(addButtonPressed()));
-    connect(user_box_, SIGNAL(returnPressed()), this, SLOT(addButtonPressed()));
+    connect(_add_button, SIGNAL(pressed()), this, SLOT(addButtonPressed()));
+    // connect(_user_box, SIGNAL(returnPressed()), this, SLOT(addButtonPressed()));
```
Comment (lunarpapillo) [inline vkconfig/vuidsearchwidget.cpp:48]: Is this an intentional removal?

If so, do we need to keep the commented-out code?

File: vkconfig/mainwindow.cpp [hunk 3]
```diff
@@ -632,55 +636,81 @@ void MainWindow::on_pushButtonAppList_clicked() {
...
+}
+
+///////////////////////////////////////////////////////////////
+// When changes are made to the layer list, it forces a reload
+// of the configuration list. This wipes everything out, so we
+// need a way to restore the currently selected item whenever
+// certain kinds of edits occur. These push/pop functions
+// accomplish that. If nothing can be found it should simply
+// leave nothing selected.
+ConfigurationListItem *MainWindow::PushCurrentItem(void) {
```
[the remark this answers is not in the document]
Comment (richard-lunarg) [reply vkconfig/mainwindow.cpp]: Originally, it was a stack... I renamed to SaveLastItem, and RestoreLastItem, with an override. They are always used in pairs, and sometimes across slots (a Qt message system), basically across function calls, so the saved value needs to be somewhere that will still be in scope, thus the class membere. This was a good critique.

File: vkconfig/vuidsearchwidget.cpp (hunk 2 above)
Comment (richard-lunarg) [reply vkconfig/vuidsearchwidget.cpp]: Removed for testing. made permanent... forgot to delete. Another good catch, and further justification for not cramming too much into one PR.

File: vkconfig/mainwindow.cpp (hunk 1 above)
Comment (richard-lunarg) [reply vkconfig/mainwindow.cpp]: Um... too much caffeine? Looks like refactoring gone bad here. Fixed for final PR.

File: vkconfig/configuration.h [hunk 4]
```diff
@@ -48,17 +48,17 @@ class Configuration {
     Configuration();
     ~Configuration();
 
-    QString name;                   // User readable display of the profile name (may contain spaces)
-                                    // This is the same as the filename, but with the .json stripped off.
-    QString file;                   // Root file name without path (by convention, no spaces and .profile suffix)
-    QString description;            // A friendly description of what this profile does
-    QByteArray setting_tree_state;  // Recall editor tree state
-    ValidationPreset preset;        // Khronos layer presets. 0 = none or user defined
+    QString _name;                   // User readable display of the profile name (may contain spaces)
```
Comment (christophe-lunarg) [inline vkconfig/configuration.h:51]: I think, it's only private members that uses "_"

**Pull request merged**